woocommerce/woocommerce
# fix: core profiler email opt in validation

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds email validation to the email marketing opt in introduced in https://github.com/woocommerce/woocommerce/pull/40869

Closes https://github.com/woocommerce/woocommerce/issues/41063.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up a WooCommerce installation or use the live branch JN instance
2. Make sure you have WC Test helper so that you can toggle the experiment assignments. Note that the status in WC Test helper doesn't update after you toggle it, but its working. Refreshing the WC Test helper page will show the updated value, but it still works even if you don't refresh.
3. Visit the core profiler 'Tell us a bit about your store' page so that the experiment is recognised in WC Test helper. Alternatively, manually add it in WC Test helper: `woocommerce_core_profiler_email_marketing_opt_in_2023_Q4_V1`
3. Toggle experiment and observe that you see the correct designs in the Business Info step of the core profiler, with email field for 'treatment' and without for 'control'



<img width="854" alt="image" src="https://github.com/woocommerce/woocommerce/assets/27843274/28deb9f6-7842-4b50-8799-29d6c4b87767">

Control:

![image](https://github.com/woocommerce/woocommerce/assets/27843274/44fdcd0a-4417-4be8-8a61-401d919db27e)

Treatment:

![image](https://github.com/woocommerce/woocommerce/assets/27843274/7b3b086d-5b62-47ae-8394-7c4ee93c0293)

##### Testing the validation

1. The validation should only apply when the opt in checkbox is visible and checked
2. The continue button should be disabled if validation fails
3. The initial validation should occur onBlur
4. Subsequent validation should happen with every keystroke so that users get fast feedback on correctness

Validation failure should look like:

<img width="668" alt="image" src="https://github.com/woocommerce/woocommerce/assets/27843274/04425fb5-b2dd-4230-8432-3d9be7137585">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
